scripts/Makefile.lib

```diff
@@ -282,9 +282,15 @@ cmd_dt_S_dtb=						\
 $(obj)/%.dtb.S: $(obj)/%.dtb
 	$(call cmd,dt_S_dtb)
 
+ifneq ($(wildcard $(PROJECT_BASE)/$(BOARD_NAME).overlay),"")
+DTS_OVERLAY = $(PROJECT_BASE)/$(BOARD_NAME).overlay
+else
+DTS_OVERLAY =
+endif
+
 quiet_cmd_dtc = DTC     $@
 cmd_dtc = echo '\#include "$(notdir $<)"' > dts/$(ARCH)/$(notdir $<)_pre_compiled ; \
-	if test -e $(PROJECT_BASE)/$(BOARD_NAME).overlay; then \
+	if test -e $(DTS_OVERLAY); then \
 		echo '\#include "$(BOARD_NAME).overlay"' >> dts/$(ARCH)/$(notdir $<)_pre_compiled ; \
```

Comment:
We use DTS_OVERLAY for the name in the test, but the #include pulls it in differently, and would ignore a setting by the user.

Perhaps use $(abspath $(PROJECT_BASE)/$(BOARD_NAME).overlay at least somewhere. Maybe have DTS_OVERLAY be user settable, and then set an absolute one to that, and use it for the #include and the dependencies.